### ISSUE TITLE
meta: Ibis expression and backend stability 

### ISSUE THREAD

**Issue by ncclementi** [ibis-project/ibis#9638]
Consolidating the https://github.com/ibis-project/ibis/issues/8996 and https://github.com/ibis-project/ibis/issues/8994 to make tracking the progress easier. 

### P1:
- [ ] https://github.com/ibis-project/ibis/issues/9000
- [ ] https://github.com/ibis-project/ibis/issues/9125 (PR POC: https://github.com/ibis-project/ibis/pull/9383)
- [ ] https://github.com/ibis-project/ibis/issues/8382
Has PR open https://github.com/ibis-project/ibis/pull/8864, but there is a new proposal to tackle this in the issue comment. 
- [ ] https://github.com/ibis-project/ibis/issues/8584

### P2 (nice to have)
- [ ] https://github.com/ibis-project/ibis/issues/8866
- [ ] https://github.com/ibis-project/ibis/issues/8727
- [ ] https://github.com/ibis-project/ibis/issues/9250
- [ ] https://github.com/ibis-project/ibis/issues/8133 

### P1
- [ ] https://github.com/ibis-project/ibis/issues/8289 (There are multiple PRs open unclear status)

### P2 (nice to have)
- [ ] https://github.com/ibis-project/ibis/issues/7280 
Has two PR aiming to tackle it https://github.com/ibis-project/ibis/pull/9096 pending on https://github.com/ibis-project/ibis/pull/9559
- [ ] https://github.com/ibis-project/ibis/issues/7133
- [ ] https://github.com/ibis-project/ibis/issues/7331
- [ ] https://github.com/ibis-project/ibis/issues/8333
